.circleci/parse_test_outputs.py

```diff
@@ -23,10 +24,18 @@ def parse_pytest_failure_output(file_path):
             match = re.match(r'^FAILED (tests/.*) - (.*): (.*)$', line)
             if match:
                 failed_count += 1
-                _, error, reason = match.groups()
-                failed_tests[reason] = failed_tests.get(reason, []) + [error]
-    for k,v in sorted(failed_tests.items(), key=lambda x:len(x[1])):
-        print(f"{len(v):4} failed because `{v[0]}` -> {k}")
+                test_name, error, reason = match.groups()
+                failed_tests[test_name] = (reason, error)
+
+
+    print("Failures:")
+    for test_name, (reason, error) in failed_tests.items():
+        print(f"{test_name} failed because `{error}` -> {reason}")
+
+    print("\nSummary:")
+    fail_counts = Counter(failed_tests.values())
+    for (reason, error), count in sorted(fail_counts.items(), key=lambda x:x[1], reverse=True):
+        print(f"{count:4} failed because of `{error}` -> {reason}")
```

Comment:
Compared to the previous logic, there is a slight difference.

- previous: each `reason` associated with a list of (possibly different) `error` and counting that list for each `reason`.
  - (I don't follow well why it is done this way, I got lazy at the end of that review)

- now: count for each pair `(reason, error)`
  -this makes more sense, so good for me